ehsm_kms_service/test/test_kms_with_rest.py

```diff
@@ -2,157 +2,156 @@
 import json
 import argparse
 import base64
+import time
+import random
+import hmac
+from hashlib import sha256
+import os
+from collections import OrderedDict
+import urllib.parse
+
+appid = '202112101919'
+appkey = '202112345678'
+
+def test_params(payload):
+    params = OrderedDict()
+
+    params["appid"] = appid
+    params["nonce"] = str(int.from_bytes(os.urandom(16), "big"))
+    params["timestamp"] = int(time.time())
+
+    query_string = urllib.parse.urlencode(params)
+    query_string += '&app_key=' + appkey
+    sign = str(base64.b64encode(hmac.new(appkey.encode('utf-8'), query_string.encode('utf-8'), digestmod=sha256).digest()),'utf-8')
+
+    params["sign"] = sign
```

Comment:
signature should contain all params including the payload.

Reply:
It will be fix.